pkg/cmd/version/version.go

```diff
@@ -15,22 +15,96 @@
 package version
 
 import (
+	"encoding/json"
 	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
 
+	"github.com/blang/semver"
+	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 )
 
 // NOTE: use go build -ldflags "-X github.com/tektoncd/cli/pkg/cmd/version.clientVersion=$(git describe)"
-var clientVersion = "dev"
+var clientVersion = devVersion
+
+const devVersion = "dev"
+const latestReleaseURL = "https://api.github.com/repos/tektoncd/cli/releases/latest"
 
 // Command returns version command
 func Command() *cobra.Command {
+	var check bool
+
 	var cmd = &cobra.Command{
 		Use:   "version",
 		Short: "Prints version information",
-		Run: func(cmd *cobra.Command, args []string) {
+		RunE: func(cmd *cobra.Command, args []string) error {
 			fmt.Fprintf(cmd.OutOrStdout(), "Client version: %s\n", clientVersion)
+
+			if !check || clientVersion == devVersion {
```

Comment:
I wonder if we should skip silently or display something like "You are running a dev version, you rock!" :man_singer: :joy: